cmd/gapid/main.go

```diff
@@ -0,0 +1,225 @@
+// Copyright (C) 2017 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"runtime"
+	"strconv"
+	"strings"
+)
+
+const versionPrefix = `version "`
+
+func main() {
+	cwd, err := filepath.Abs(filepath.Dir(os.Args[0]))
+	if err != nil {
+		fmt.Println(err)
+		os.Exit(1)
+	}
+
+	cwd, err = filepath.EvalSymlinks(cwd)
+	if err != nil {
+		fmt.Println(err)
+		os.Exit(1)
+	}
+
+	args := os.Args[1:]
+	gapicArgs := make([]string, 0, len(args))
+	vm := ""
+	vmargs := make([]string, 0)
+	help := false
+	console := false
+
+	for i := 0; i < len(args); i++ {
+		switch {
+		case args[i] == "--vm" && i < len(args)-1:
+			i = i + 1
+			vm = args[i]
+		case args[i] == "--vmarg" && i < len(args)-1:
+			i = i + 1
+			vmargs = append(vmargs, args[i])
+		case args[i] == "--console":
+			console = true
+			createConsole()
+		default:
+			help = help || args[i] == "--help"
+			gapicArgs = append(gapicArgs, args[i])
+		}
+	}
+
+	if help && !console {
+		console = true
+		createConsole()
+	}
+
+	java, noVM := locateVM(cwd, vm, console)
+	gapic, noGAPIC := locateGapic(cwd)
+
+	failed := noVM || noGAPIC
+	if !failed {
+		fmt.Println("Starting", java, gapic)
+		if runtime.GOOS == "darwin" {
+			vmargs = append(vmargs, "-XstartOnFirstThread")
+		}
+
+		vmargs = append(vmargs, "-jar", gapic)
+		cmd := exec.Command(java, append(vmargs, gapicArgs...)...)
+		cmd.Stdin = os.Stdin
+		cmd.Stdout = os.Stdout
+		cmd.Stderr = os.Stderr
+
+		cmd.Env = append(os.Environ(), "GAPID="+cwd)
+		if runtime.GOOS == "linux" {
+			cmd.Env = append(cmd.Env, "SWT_GTK3=0", "LIBOVERLAY_SCROLLBAR=0")
+		}
+
+		if err := cmd.Run(); err != nil {
+			failed = true
+			if _, ok := err.(*exec.ExitError); !ok {
+				fmt.Println("Failed to start GAPIC:", err)
+			}
+		}
+	}
+
+	if help {
+		fmt.Println()
+		fmt.Println("Launcher Flags:")
+		fmt.Println(" --vm              Path to the JVM to use")
+		fmt.Println(" --vmarg           Extra argument for the JVM (repeatable)")
+		fmt.Println(" --console         Run GAPID inside a terminal console")
+	}
+
+	if console && runtime.GOOS == "windows" {
+		fmt.Println()
+		fmt.Println("Press enter to continue")
+		os.Stdin.Read(make([]byte, 1))
+	}
+
+	if failed {
+		os.Exit(1)
+	}
+}
+
+func locateVM(cwd, vm string, console bool) (string, bool) {
```

Comment:
It's more idiomatic to return value, error. Return the error message and have the caller print it.
In the few cases where there are bool return values to denote success/failure, true represents success. Using `error` is unambiguous.

Reply:
Done.